pulsar-broker/src/main/java/com/yahoo/pulsar/broker/service/BrokerService.java

```diff
@@ -418,6 +431,15 @@ public PulsarClient getReplicationClient(String cluster) {
     }
 
     private CompletableFuture<Topic> createPersistentTopic(final String topic) throws RuntimeException {
+        try {
+            return createPersistentTopic(topic, false);
+        } catch (TooManyRequestsException e) {
+            return FutureUtil.failedFuture(e);
+        }
+    }
+
+    private CompletableFuture<Topic> createPersistentTopic(final String topic, boolean failOnTooManyConcurrentTopics)
```

Comment:
```
void createPersistentTopic(String topic, CompletableFuture<Topic> topicFuture) {
   // To be called when we are ready to load the topic
}
```
Does this method throttle again and puts in the queue again or it just fails the future if semaphore doesn't have any additional permits?